x/authority/keeper/authorization_list.go

```diff
@@ -0,0 +1,30 @@
+package keeper
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/zeta-chain/zetacore/x/authority/types"
+)
+
+// SetAuthorizationList sets the authorization list to the store.It returns an error if the list is invalid.
+func (k Keeper) SetAuthorizationList(ctx sdk.Context, list types.AuthorizationList) error {
```

Comment:
It is easier to keep Set functions non-opiniated as a low level function. This is the caller that should ensure the type is valid

This is what we have with all other types

Reply:
If we do it, we should address in a separate PR for consistency and I think this is not necessary, the setter/getter are the low-level API for the store to save/retrieve data, this level should not be aware of what represent the data. The validation should always be called in the high-level logic. Validating genesis, validating message, message processing